common/shared.c

```diff
@@ -1136,6 +1136,38 @@ void init_test(struct ft_opts *opts, char *test_name, size_t test_name_len)
 		opts->iterations = size_to_count(opts->transfer_size);
 }
 
+static int ft_inject_progress(uint64_t total)
+{
+	struct fi_cq_err_entry comp;
+	int ret;
+
+	if (txcq) {
+		if (opts.comp_method == FT_COMP_SREAD)
+			ret = fi_cq_sread(txcq, &comp, 1, NULL, 0);
+		else
+			ret = fi_cq_read(txcq, &comp, 1);
+		/* Even though we're progressing inject we may get a completion from
+		 * a previous non-inject operation */
+		if (ret > 0)
+			tx_cq_cntr++;
+	} else if (txcntr) {
+		ret = fi_cntr_wait(txcntr, total, -1);
```

Comment:
The timeout in FT_POST is set to 0 and that should work. But if this function is called from other places when timeout could be -1, we would have a problem. For inject progress, attempting to wait on the counter just once (with zero timeout) should be enough, right?